### ISSUE TITLE
Why are you using Node

### ISSUE THREAD

**Issue by ghost** [mklement0/perli#2]
This is not JavaScript

**Comment by mklement0** [comment]
Ah, I see - it's a good thing I didn't go with my initial choice of name, "javascripti". 😜

The truth is that I was already familiar with npm as a capable cross-platform package repository and manager that can also handle CLIs, and I didn't want to spend the time learning how to publish to CPAN too.

Do you think limiting the availability to npm makes many people miss out on / not bother with `perli`?

**Comment by cup** [comment]
its just... weird

technically its fine, but personally i think the whole Node project is ridiculous

javascript isnt and wasnt ever meant to be a systems programming language, it just seems like the whole project is people trying to ram a square into a circle

then you get projects like Electron and *shudder*

i think youd be much better off on https://metacpan.org

**Comment by smac89** [comment]
Part of the fun of programming is being able to discover new ways to use existing tools rather than re-inventing the wheel or resorting to manually doing what you need.

NPM is a package manager designed to work for JavaScript projects, but it doesn't mean it only has to work for JavaScript.

Same reason why most GTK python projects use meson as a package manager even though meson does not list python as one of the languages it supports. They use it because meson is a more configurable option than `setup.py`, for cross-platform development

**Comment by ernstki** [comment]
Agreed, in principle, and I was also surprised at this, given the **relative**—_relative_!—ease with which you're able to get CPAN packages on any system with Perl already installed.

But this made me laugh at my own (negative) assumptions about npm, mostly because I don't really _use_ it as a developer, just an end user.

If _I_ had experience somewhere else, I'm not sure I'd go willingly, either, into the dark pit of briny tentacles that is Perl packaging.

Stay weird, everyone. `:)`
